### DISCUSSION TITLE
Custom Human-in-the-loop inflates latency metrics for Langgraph run

### DISCUSSION THREAD

**Discussion by runirudh** [langchain-ai/langsmith-sdk#731] [Q&A]
I have a few tools and nodes in my LangGraph that implement human in the loop. Tracing in LangSmith, the latency for my graph is inflated due to the pausing of graph execution at certain points in the graph. Currently I pause the graph using `threading.Event()`.

Is there a way to specify a list of tools that don't contribute to the latency, or something simpler like using memory + checkpointing  that does not distort the latency during tracing.  I'm not currently using checkpointing as my backend is mongodb which isn't supported out of the box.

**Comment by hinthornw** [comment]
Great question! There isn't a way to pause time within a span, and I'm not sure it's a feature we plan to add.

Once the checkpointing API solidifies we will document (plus provide a managed option) how to make a custom checkpointer, which should resolve this.

For now, one thing you could do is create a nested run to handle the waiting, then at least you can infer the actual tool time from the difference between the two
